avogadro/qtgui/rwmolecule.h

```diff
@@ -744,6 +758,19 @@ inline std::pair<Index, Index> RWMolecule::bondPair(Index bondId) const
   return m_molecule.bondPair(bondId);
 }
 
+inline void RWMolecule::beginDraw(const QString& undoName)
+{
+  m_undoStack.beginMacro(undoName);
+}
+
+inline void RWMolecule::endDraw()
+{
+  m_undoStack.endMacro();
+  // Let's cover all possibilities
+  emitChanged(Molecule::Atoms | Molecule::Bonds |
+              Molecule::Added | Molecule::Removed);
+}
+
 inline void RWMolecule::setInteractive(bool b)
```

Comment:
Can't you just modify this, e.g.

setInteractive(bool b, const QString &undoName = "Draw")
{ 
  if (b) m_undoStack.beginMacro(undoName);
  else // ..